File: Client/app/src/main/java/com/example/client/networking/responses/UserResponse.java
package com.example.client.networking.responses;

import com.google.gson.Gson;

public class UserResponse {
    private int id;
    private String email;
    private String name;
    private String pictureUrl;
    private String roleName;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public void setPictureUrl(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }

    public String toJson() {
        return new Gson().toJson(this);
    }

    public static UserResponse fromJson(String json) {
        return new Gson().fromJson(json, UserResponse.class);
    }
}

File: Server/Server/DTO/UserResponse.cs
namespace Server.DTO
{
    public class UserResponse
    {
        public int Id { get; set; }
        public string Email { get; set; }
        public string Name { get; set; }
        public string PictureUrl { get; set; }
        public string RoleName { get; set; }

        public UserResponse()
        {
            Id = 0;
            Email = string.Empty;
            Name = string.Empty;
            PictureUrl = string.Empty;
            RoleName = string.Empty;
        }
    }
}
